### PR TITLE
[docs] Describe requestOptions.isAjax

### LINKED ISSUE #5562: Add 'req.isAjax' option description
* Documentation - https://devexpress.github.io/testcafe/documentation/reference/test-api/requestmock/respond.html#requestoptions

* Typings - https://github.com/DevExpress/testcafe/blob/1e30f9de72e27a2afe10b1fdec317456bffb9723/ts-defs-src/test-api/request-hook.d.ts#L171

### PR DESCRIPTION
**Pull request by DIRECTcut**
fixes #5562

### CODE REVIEW

File: docs/articles/templates/intercept-http-requests/requestoptions-object.md [hunk 1]
```diff
@@ -1,15 +1,16 @@
...
+`hostname`    | String  | The alias for the host.
+`host`        | String  | The domain name or IP address of the server to issue the request to. Default: *localhost*.
+`port`        | Number  | The port of the remote server. Default: *80*.
+`path`        | String  | The request path. Should include query string if any. For example, *'/index.html?page=12'*. An exception is thrown when the request path contains illegal characters (for example, spaces). Default: *'/'*.
+`method`      | String  | The string specifying the HTTP request method. Default: *'GET'*.
+`credentials` | Object  | Credentials that were used for authentication in the current session with NTLM or Basic authentication. For HTTP Basic authentication, these are `username` and `password`. NTLM authentication additionally specifies `workstation` and `domain`. See [HTTP Authentication](/testcafe/documentation/guides/advanced-guides/authentication.html#http-authentication).
+`proxy`       | Object  | If a proxy is used, the property contains information about its `host`, `hostname`, `port`, `proxyAuth`, `authHeader` and `bypassRules`.
+`isAjax`      | Boolean | Specifies whether the request is AJAX.
```
Comment (miherlosev) [inline docs/articles/templates/intercept-http-requests/requestoptions-object.md:16]: `Specifies whether the request is AJAX request (xhr or fetch).`

**Review by miherlosev — CHANGES_REQUESTED** [review]

File: docs/articles/templates/intercept-http-requests/requestoptions-object.md (hunk 1 above)
Comment (DIRECTcut) [reply docs/articles/templates/intercept-http-requests/requestoptions-object.md]: Edited

**Review by miherlosev — APPROVED** [review]

File: docs/articles/templates/intercept-http-requests/requestoptions-object.md [hunk 2]
```diff
@@ -1,15 +1,16 @@
...
-`host`     | String | The domain name or IP address of the server to issue the request to. Default: *localhost*.
-`port`     | Number | The port of the remote server. Default: *80*.
-`path`     | String | The request path. Should include query string if any. E.G. *'/index.html?page=12'*. An exception is thrown when the request path contains illegal characters. Currently, only spaces are rejected but that may change in the future. Default: *'/'*.
-`method`   | String | The string specifying the HTTP request method. Default: *'GET'*.
-`credentials` | Object | Credentials that were used for authentication in the current session using NTLM or Basic authentication. For HTTP Basic authentication, these are `username` and `password`. NTLM authentication additionally specifies `workstation` and `domain`. See [HTTP Authentication](/testcafe/documentation/guides/advanced-guides/authentication.html#http-authentication).
-`proxy`       | Object | If a proxy is used, the property contains information about its `host`, `hostname`, `port`, `proxyAuth`, `authHeader` and `bypassRules`.
\ No newline at end of file
+`url`         | String  | The URL to which to send the request.
```
Comment (titerman) [inline docs/articles/templates/intercept-http-requests/requestoptions-object.md:7]: that's a mouthful. Would this work:
```suggestion
`url`         | String  | The URL of the resource.
```

File: docs/articles/templates/intercept-http-requests/requestoptions-object.md [hunk 3]
```diff
@@ -1,15 +1,16 @@
...
+`protocol`    | String  | The protocol to use. Default: *http:*.
```
Comment (titerman) [inline docs/articles/templates/intercept-http-requests/requestoptions-object.md:8]: Why not:
```suggestion
`protocol`    | String  | The request protocol. Default: *http:*.
```

File: docs/articles/templates/intercept-http-requests/requestoptions-object.md [hunk 4]
```diff
@@ -1,15 +1,16 @@
...
+`hostname`    | String  | The alias for the host.
+`host`        | String  | The domain name or IP address of the server to issue the request to. Default: *localhost*.
```
Comment (titerman) [inline docs/articles/templates/intercept-http-requests/requestoptions-object.md:10]: Honest question: what's the difference b/w `host` and `hostname`? I don't quite understand

File: docs/articles/templates/intercept-http-requests/requestoptions-object.md [hunk 5]
```diff
@@ -1,15 +1,16 @@
...
+`port`        | Number  | The port of the remote server. Default: *80*.
+`path`        | String  | The request path. Should include query string if any. For example, *'/index.html?page=12'*. An exception is thrown when the request path contains illegal characters (for example, spaces). Default: *'/'*.
+`method`      | String  | The string specifying the HTTP request method. Default: *'GET'*.
```
Comment (titerman) [inline docs/articles/templates/intercept-http-requests/requestoptions-object.md:13]:
```suggestion
`method`      | String  | The HTTP request method. Default: *'GET'*.
```

File: ts-defs-src/test-api/request-hook.d.ts [hunk 6]
```diff
@@ -203,6 +203,10 @@ interface RequestOptions {
      * `proxyAuth`, `authHeader` and `bypassRules`.
      */
     proxy: Record<string, unknown>;
+    /**
+     * Specifies whether the request is AJAX (xhr or fetch).
```
Comment (titerman) [inline ts-defs-src/test-api/request-hook.d.ts:207]: AJAX is not an adjective :(

**Review by titerman — COMMENTED** [review]
A couple of suggestions

File: docs/articles/templates/intercept-http-requests/requestoptions-object.md (hunk 2 above)
Comment (DIRECTcut) [reply docs/articles/templates/intercept-http-requests/requestoptions-object.md:7]: Shorter and works. Applied

File: docs/articles/templates/intercept-http-requests/requestoptions-object.md (hunk 3 above)
Comment (DIRECTcut) [reply docs/articles/templates/intercept-http-requests/requestoptions-object.md:8]: Seems same to me

File: docs/articles/templates/intercept-http-requests/requestoptions-object.md (hunk 4 above)
Comment (DIRECTcut) [reply docs/articles/templates/intercept-http-requests/requestoptions-object.md:10]: No clue
@miherlosev ?

File: ts-defs-src/test-api/request-hook.d.ts (hunk 6 above)
Comment (DIRECTcut) [reply ts-defs-src/test-api/request-hook.d.ts:207]: That's a shame(

> Specifies whether the request is an AJAX request (xhr or fetch).

File: docs/articles/templates/intercept-http-requests/requestoptions-object.md (hunk 4 above)
Comment (miherlosev) [reply docs/articles/templates/intercept-http-requests/requestoptions-object.md:10]: https://nodejs.org/api/url.html#url_url_host

File: ts-defs-src/test-api/request-hook.d.ts (hunk 6 above)
Comment (miherlosev) [reply ts-defs-src/test-api/request-hook.d.ts]: https://docs.microsoft.com/en-us/dotnet/api/system.web.mvc.ajaxrequestextensions.isajaxrequest?view=aspnet-mvc-5.2

**Review by VasilyStrelyaev — APPROVED** [review]

File: docs/articles/templates/intercept-http-requests/requestoptions-object.md (hunk 3 above)
Comment (arminal) [reply docs/articles/templates/intercept-http-requests/requestoptions-object.md:8]: I agree with this correction.

File: docs/articles/templates/intercept-http-requests/requestoptions-object.md [hunk 7]
```diff
@@ -1,15 +1,16 @@
...
-`credentials` | Object | Credentials that were used for authentication in the current session using NTLM or Basic authentication. For HTTP Basic authentication, these are `username` and `password`. NTLM authentication additionally specifies `workstation` and `domain`. See [HTTP Authentication](/testcafe/documentation/guides/advanced-guides/authentication.html#http-authentication).
-`proxy`       | Object | If a proxy is used, the property contains information about its `host`, `hostname`, `port`, `proxyAuth`, `authHeader` and `bypassRules`.
\ No newline at end of file
+`url`         | String  | The URL of the resource.
+`protocol`    | String  | The protocol to use. Default: *http:*.
+`hostname`    | String  | The alias for the host.
+`host`        | String  | The domain name or IP address of the server to issue the request to. Default: *localhost*.
+`port`        | Number  | The port of the remote server. Default: *80*.
+`path`        | String  | The request path. Should include query string if any. For example, *'/index.html?page=12'*. An exception is thrown when the request path contains illegal characters (for example, spaces). Default: *'/'*.
```
Comment (arminal) [inline docs/articles/templates/intercept-http-requests/requestoptions-object.md:12]:
```suggestion
`path`        | String  | The request path. Should include the query string if any. For example, *'/index.html?page=12'*. An exception is thrown when the request path contains illegal characters (e.g.,, spaces). Default: *'/'*.
```

File: docs/articles/templates/intercept-http-requests/requestoptions-object.md (hunk 4 above)
Comment (titerman) [reply docs/articles/templates/intercept-http-requests/requestoptions-object.md:10]: The difference is not obvious from the documentation. Here's what I suggest:

```suggestion
`hostname`    | String  | The domain name or IP address of the destination server. Default: *localhost*.
`host`        | String  | The domain name or IP address of the destination server, including the listening port. Default: *localhost:80*.
```

Comment (titerman) [reply docs/articles/templates/intercept-http-requests/requestoptions-object.md:10]: @DIRECTcut @miherlosev

File: docs/articles/templates/intercept-http-requests/requestoptions-object.md (hunk 7 above)
Comment (DIRECTcut) [reply docs/articles/templates/intercept-http-requests/requestoptions-object.md]: applied

**Review by titerman — APPROVED** [review]

**Pull request merged**